pkg/ccl/backupccl/backup_test.go

```diff
@@ -9678,6 +9711,7 @@ func TestBackupNoOverwriteCheckpoint(t *testing.T) {
 		},
 	}
 	params.ServerArgs.Knobs = knobs
+	params.ServerArgs.DisableDefaultTestTenant = true
```

Comment:
This test has at 2 problems:

1) It accesses `tc.Server(0).ExecutorConfig().(sql.ExecutorConfig)` in order to look up the userfile we are writing to.  Locally, I've added a `tc.TenantOrServer(0)` method that returns the correct TestTenantInterface which can be used to resolve the issue.

2) It looks like when we start the test tenant, we don't copy over the  testing knobs.  If I locally add some code to copy over a few of the relevant testing knobs, the test passes.